internal/util/setters/setters.go

```diff
@@ -29,30 +31,45 @@ import (
 	"sigs.k8s.io/kustomize/kyaml/openapi"
 	"sigs.k8s.io/kustomize/kyaml/pathutil"
 	"sigs.k8s.io/kustomize/kyaml/setters"
+	"sigs.k8s.io/kustomize/kyaml/setters2"
 	"sigs.k8s.io/kustomize/kyaml/setters2/settersutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
 const (
 	GcloudProject       = "gcloud.core.project"
 	GcloudProjectNumber = "gcloud.project.projectNumber"
 )
 
-func PerformAutoSetters(path string) error {
-	// auto-fill setters from environment
-	if err := SetEnvAutoSetters(path); err != nil {
+type AutoSet struct {
+	Writer     io.Writer
+	TargetPath string
+	ParentPath string
+}
+
+// PerformAutoSetters auto-fills the setter values from the local environment
```

Comment:
Can we explain in some more detail here what is happening? In particular how we handle different types of autosetters and how we handle subpackage.

Reply:
Will do.